File: go/httpmw/conformance_test.go
package httpmw_test

import (
	"net/http"
	"testing"
	"time"

	"github.com/sns45/anyonce/go/anyonce"
	"github.com/sns45/anyonce/go/conformance"
	"github.com/sns45/anyonce/go/conformance/fixture"
	"github.com/sns45/anyonce/go/httpmw"
	"github.com/sns45/anyonce/go/store/memory"
)

func TestConformance(t *testing.T) {
	t.Run("REQ-HTTP-18: every core and profile vector passes through httpmw with the memory store", func(t *testing.T) {
		f := fixture.New()
		mw := httpmw.New(memory.New(), httpmw.Options{Required: true, Policy: anyonce.Policy{TTL: 2 * time.Second}})
		mux := http.NewServeMux()
		mux.Handle("POST /reset", f.Handler())
		mux.Handle("/", mw.Handler(f.Handler()))
		summary := conformance.Run(t, mux, conformance.Options{Capabilities: []string{"short-ttl"}})
		if summary.Passed != len(summary.Results) || len(summary.Results) != 20 {
			t.Fatalf("passed %d of %d: %+v", summary.Passed, len(summary.Results), summary.Results)
		}
	})
}

File: packages/core/test/http/conformance.test.ts
import { describe, expect, test } from 'bun:test';
import { runConformance } from '@anyonce/conformance';
import { createFixtureApp } from '@anyonce/fixture-hono';
import { CORE_IDS, PROFILE_IDS } from '../../../conformance/test/catalog';
import { withIdempotency } from '../../src/http';
import { MemoryStore } from '../../src/memory';

describe('withIdempotency conformance', () => {
  test('REQ-HTTP-17: every core and profile vector passes through withIdempotency with the memory store', async () => {
    const app = createFixtureApp();
    const handler = withIdempotency(app.fetch, {
      store: new MemoryStore(),
      required: true,
      ttlMs: 2000,
      skip: (req) => new URL(req.url).pathname === '/reset',
    });
    const { summary, report } = await runConformance({
      target: handler,
      capabilities: ['short-ttl'],
      report: 'markdown',
    });
    const notPassing = summary.results
      .filter((r) => r.status !== 'pass')
      .map((r) => `${r.id}: ${r.status}`);
    expect(notPassing, report).toEqual([]);
    expect(summary.results).toHaveLength(CORE_IDS.length + PROFILE_IDS.length);
    expect(summary.passed).toBe(CORE_IDS.length + PROFILE_IDS.length);
  }, 60_000);
});
